diff --git a/tensorflow/core/BUILD b/tensorflow/core/BUILD
index 8fd8f3b25..a8edb1d75 100644
--- a/tensorflow/core/BUILD
+++ b/tensorflow/core/BUILD
@@ -675,6 +675,7 @@ cc_library(
 cc_library(
     name = "all_kernels",
     visibility = ["//visibility:public"],
+    copts = tf_copts(),
     deps = [
         "//tensorflow/core/kernels:array",
         "//tensorflow/core/kernels:bincount_op",
@@ -1384,7 +1385,7 @@ tf_cuda_library(
         "util/tensor_slice_set.h",
         "util/tensor_slice_util.h",
     ],
-    copts = tf_copts(),
+    copts = tf_copts() + ["-g"],
     linkopts = select({
         "//tensorflow:freebsd": [],
         "//conditions:default": ["-ldl"],
@@ -1486,7 +1487,7 @@ tf_cuda_library(
             "common_runtime/gpu_device_context.h",
         ],
     ),
-    copts = tf_copts(),
+    copts = tf_copts() + ["-g"],
     deps = [
                ":framework",
                ":framework_internal",
@@ -1600,7 +1601,7 @@ tf_cuda_library(
         "common_runtime/gpu/pool_allocator.h",
         "common_runtime/gpu/process_state.h",
     ],
-    copts = tf_copts(),
+    copts = tf_copts() + ["-g"],
     linkstatic = 1,
     deps = [
         ":core_cpu",
@@ -1675,7 +1676,7 @@ cc_library(
         "common_runtime/rpc_device/rpc_device_context.h",
         "common_runtime/rpc_device/threadpiscine_device.h",
     ],
-    copts = tf_copts() + ["-fexceptions"],
+    copts = tf_copts() + ["-fexceptions", "-g"],
     linkstatic = 1,
     deps = [
         ":core_cpu",
diff --git a/tensorflow/core/distributed_runtime/BUILD b/tensorflow/core/distributed_runtime/BUILD
index e26741465..af412cc22 100644
--- a/tensorflow/core/distributed_runtime/BUILD
+++ b/tensorflow/core/distributed_runtime/BUILD
@@ -55,6 +55,7 @@ cc_library(
         "//tensorflow/core:proto_text",
         "//tensorflow/core:worker_proto_cc",
     ],
+    copts = ["-g"],
 )
 
 cc_test(
@@ -67,12 +68,14 @@ cc_test(
         "//tensorflow/core:test",
         "//tensorflow/core:test_main",
     ],
+    copts = ["-g"],
 )
 
 cc_library(
     name = "worker_env",
     hdrs = ["worker_env.h"],
     deps = [],
+    copts = ["-g"],
 )
 
 cc_library(
@@ -90,6 +93,7 @@ cc_library(
         "//tensorflow/core:lib",
         "//tensorflow/core:worker_proto_cc",
     ],
+    copts = ["-g"],
 )
 
 cc_library(
@@ -105,6 +109,7 @@ cc_library(
         "//tensorflow/core:core_cpu_internal",
         "//tensorflow/core:lib_internal",
     ],
+    copts = ["-g"],
 )
 
 cc_library(
@@ -114,6 +119,7 @@ cc_library(
     deps = [
         "//tensorflow/core:lib",
     ],
+    copts = ["-g"],
 )
 
 cc_test(
@@ -125,6 +131,7 @@ cc_test(
         "//tensorflow/core:test",
         "//tensorflow/core:test_main",
     ],
+    copts = ["-g"],
 )
 
 cc_test(
@@ -142,12 +149,14 @@ cc_test(
         "//tensorflow/core:test_main",
         "//tensorflow/core:worker_proto_cc",
     ],
+    copts = ["-g"],
 )
 
 cc_library(
     name = "worker_cache",
     hdrs = ["worker_cache.h"],
     deps = ["//tensorflow/core:protos_all_cc"],
+    copts = ["-g"],
 )
 
 cc_library(
@@ -161,6 +170,7 @@ cc_library(
         "//tensorflow/core:lib",
         "//tensorflow/core:worker_proto_cc",
     ],
+    copts = ["-g"],
 )
 
 cc_library(
@@ -171,6 +181,7 @@ cc_library(
         "//tensorflow/core:lib",
         "//tensorflow/core:master_proto_cc",
     ],
+    copts = ["-g"],
 )
 
 cc_library(
@@ -192,6 +203,7 @@ cc_library(
         "//tensorflow/core:master_proto_cc",
         "//tensorflow/core:worker_proto_cc",
     ],
+    copts = ["-g"],
 )
 
 cc_library(
@@ -213,6 +225,7 @@ cc_library(
         "//tensorflow/core:master_proto_cc",
         "//tensorflow/core:protos_all_cc",
     ],
+    copts = ["-g"],
 )
 
 cc_library(
@@ -224,6 +237,7 @@ cc_library(
         ":master_interface",
         "//tensorflow/core:lib",
     ],
+    copts = ["-g"],
 )
 
 cc_library(
@@ -233,6 +247,7 @@ cc_library(
     deps = [
         "//tensorflow/core:framework",
     ],
+    copts = ["-g"],
 )
 
 cc_library(
@@ -244,6 +259,7 @@ cc_library(
         "//tensorflow/core:framework",
         "//tensorflow/core:tensorflow_opensource",
     ],
+    copts = ["-g"],
 )
 
 cc_library(
@@ -260,11 +276,13 @@ cc_library(
         "//tensorflow/core:framework_internal",
         "//tensorflow/core:lib",
     ],
+    copts = ["-g"],
 )
 
 cc_library(
     name = "master_env",
     hdrs = ["master_env.h"],
+    copts = ["-g"],
 )
 
 cc_library(
@@ -281,6 +299,7 @@ cc_library(
         "//tensorflow/core:protos_all_cc",
         "//tensorflow/core:worker_proto_cc",
     ],
+    copts = ["-g"],
 )
 
 cc_library(
@@ -295,6 +314,7 @@ cc_library(
         "//tensorflow/core:lib",
         "//tensorflow/core:worker_proto_cc",
     ],
+    copts = ["-g"],
 )
 
 cc_library(
@@ -306,6 +326,7 @@ cc_library(
         "//tensorflow/core:lib",
         "//tensorflow/core:protos_all_cc",
     ],
+    copts = ["-g"],
 )
 
 cc_library(
@@ -318,6 +339,7 @@ cc_library(
         "//tensorflow/core:lib",
         "//tensorflow/core:protos_all_cc",
     ],
+    copts = ["-g"],
 )
 
 # TODO(mrry): Move executor_test.cc to ../common_runtime when once it no longer depends
diff --git a/tensorflow/core/distributed_runtime/zrpc/zrpc.bzl b/tensorflow/core/distributed_runtime/zrpc/zrpc.bzl
index af8aeca36..4bd0a5261 100644
--- a/tensorflow/core/distributed_runtime/zrpc/zrpc.bzl
+++ b/tensorflow/core/distributed_runtime/zrpc/zrpc.bzl
@@ -10,7 +10,7 @@ load(
 )
 
 def zrpc_copts():
-    return ([]
+    return (["-g"]
     + if_cuda(["-DGOOGLE_CUDA=1"])
     + if_mkl(["-DINTEL_MKL=1"])
     + if_android_arm(["-mfpu=neon"])
