diff --git a/tensorflow/workspace.bzl b/tensorflow/workspace.bzl
index a64afb3c5..b27660fce 100644
--- a/tensorflow/workspace.bzl
+++ b/tensorflow/workspace.bzl
@@ -421,7 +421,7 @@ def tf_workspace(path_prefix = "", tf_repo_name = ""):
 
   native.new_local_repository(
       name = "zeromq",
-      path = "/usr",
+      path = "spack-packages/",
       build_file = str(Label("//third_party:zeromq.BUILD")),
   )
 
